Editor/Window/ElementLocalizer.cs

```diff
@@ -16,12 +17,37 @@ public ElementLocalizer(VisualElement root)
             _root = root;
         }
 
+        public string GetText(string textKey)
+        {
+            return _textProvider.Get(textKey);
+        }
+
         public void SetElementText(string elementName, string textKey)
+        {
+            var text = _textProvider.Get(textKey);
+            SetText(elementName, text);
+        }
+
+        public void SetElementText(string elementName, string textKey, Dictionary<string, string> wordReplacements)
+        {
+            var text = wordReplacements.Aggregate(_textProvider.Get(textKey), (result, next) => result.Replace($"[{next.Key}]", next.Value));
+            SetText(elementName, text);
+        }
+
+        private void SetText(string elementName, string text)
         {
             var element = _root.Q<TextElement>(elementName);
             if (element != null)
             {
-                element.text = _textProvider.Get(textKey);
+                element.text = text;
+            }
+            else
+            {
+                var foldout = _root.Q<Foldout>(elementName);
+                if (foldout != null)
+                {
+                    foldout.text = text;
+                }
```

Comment:
This seems like a weird assumption that if the element is not a `TextElement`, it will be a `Foldout`. Is that applicable to all our occurrences? Seems like we may want to use to setText on a third type of element, and this code would become a bunch of if-elses at that point. Is there a better way for this?

Reply:
Currently the only element we use that does not extend from `TextElement` is `Foldout`. We have attempted generics but not been able to settle on an elegant solution. The only other option I can think about is wrapping Foldout and essentially reexporting it extending from `TextElement`

Reply:
Going to ignore for now since this seems minor